webapp/src/Controller/API/RunController.php

```diff
@@ -152,8 +152,10 @@ protected function getQueryBuilder(Request $request): QueryBuilder
                 ->setParameter('judging_id', $request->query->get('judging_id'));
         }
 
-        if ($request->query->has('limit')) {
-            $queryBuilder->setMaxResults($request->query->getInt('limit'));
+	if ($request->query->has('limit')) {
+	    $limit = $request->query->getInt('limit');
+	    $limit = ($limit>0 ? $limit : 0);
```

Comment:
alternatively error out when `$limit < 0`?

Reply:
That might be even better it explains why you would get 0 results.